File: src/main/java/com/micabytes/ink/StoryWrapper.java
package com.micabytes.ink;

import org.jetbrains.annotations.NonNls;

import java.io.InputStream;

/** StoryWrapper is the interface to the class used to convert file IDs/file names to and
 * InputStream.
 */
public interface StoryWrapper {
  InputStream getStream(String fileId);

  Object getStoryObject(String objId);

  StoryInterrupt getInterrupt(String s);

  void logDebug(String m);

  void logError(@NonNls String m);

  void logException(Exception e);
}

File: src/main/kotlin/com/micabytes/ink/StoryWrapper.kt
package com.micabytes.ink

import java.io.InputStream

/// StoryWrapper is the interface to the class used to convert file IDs/file names to and InputStream.
interface StoryWrapper {
  fun getStream(fileId: String): InputStream
  fun getStoryObject(objId: String): Any
  fun getInterrupt(s: String): StoryInterrupt
  fun resolveTag(t: String)
  fun logDebug(m: String)
  fun logError(m: String)
  fun logException(e: Exception)
}
